src/Utilities.sol

```diff
@@ -252,7 +256,7 @@ library PolyLib {
     }
 
     function evalAtOne(UniPoly memory poly, uint256 mod) public pure returns (uint256 result) {
-        for (uint256 i = 0; i < poly.coeffs.length; i++) {
+        for (uint256 i; i < poly.coeffs.length; i++) {
```

Comment:
Why not using approach with `unchecked` incrementing of the index here?

Reply:
Actually, solidity version 0.8 constituted 'underflow/overflow' checks which obviously costs some amount of gas and they are damn important. `unchecked` kind of discards these checks and then the code inside the 'unchecked' block becomes prone to these underflow/overflow bugs. Thus, we really can't used 'unchecked' everywhere..
Now, coming back to your question, There were mainly two types of 'for' loops used in this solidity file:

```
// 1. This one has a fixed limit of 32, which makes me pretty sure
// that 'i' will never overflow
for (uint i; i < 32;){
    // code
    unchecked {
        ++i;
    }
}

// 2. But this one doesn't have a fixed limit, 
// like we can't say what size of array the user can provide. 
// That's why, really doubtful in here and avoided making any changes
for (uint i; i < poly.coeffs.length; i++){
    // code
}   
```

Still, I feel that there won't be any problem with `++i`, but still if you really assure me that poly.coeffs.length doesn't cause any bugs in the code due to the addition of `unchecked` block, then it be worth making these changes here too!

This answer also aligns well with your previous question -> ' Can we populate this approach to the rest of for loops in the codebase?'
Still like to say that, Yes we can inscribe this approach in rest of the for loops in this codebase, but need to be more careful where 'underflow/overflow' checks are really needed and where they aren't

Hence, I will make these changes once you allow for it. But do make sure to double-check them

Reply:
Yeah, that makes sense. Let me ask @huitseeker's opinion here.

Now offhand, I couldn't imagine the situation where particular loop counter with unfixed limits may overflow. Otherwise, it would mean that we need to iterate over extremely large array of uint256s (we use them most of all) which I believe will cause OutOfGasException.

Reply:
True..I agree with that. Iterating over large arrays might cause OutOfGasException. But still many repositories and projects kind of avoids using unchecked in many scenarios. Anyways, yeah we must also say that those repositories and projects do lack these gas optimization fixes too.
However, even I feel that there won't be any problem in making these changes even with the arrays. Let's wait for @huitseeker opinion